doc/source/whatsnew/v0.18.0.txt

```diff
@@ -502,4 +502,6 @@ Bug Fixes
 
 - Bug in ``.to_csv`` ignoring formatting parameters ``decimal``, ``na_rep``, ``float_format`` for float indexes (:issue:`11553`)
 
+- Bug in ``Int64Index`` preventing the use of the modulo operator (:issue:`9244`)
```

Comment:
this is very narrow what you are saying. pls test & expand (this should work for numeric types)